### PR TITLE
fix(cursor): cursor.count not respecting parent readPreference

### PR DESCRIPTION
**Pull request by rweinberger**
Fix wrong placement and typo in options argument in
cursor.s.topology.command, and test that specified readPreference
is passed in to that call.

Fixes NODE-1511

### CODE REVIEW

File: test/functional/cursor_tests.js [hunk 1]
```diff
@@ -4501,4 +4502,34 @@ describe('Cursor', function() {
 
     testTransformStream(config, done);
   });
+
+  it('should apply parent read preference to count command', function(done) {
```
Comment (kvwalker) [inline test/functional/cursor_tests.js]: Does this test pass with a replicaset? It's not passing for me locally.

Comment (rweinberger) [reply test/functional/cursor_tests.js]: How can we imitate your local environment? We're not sure why it would be failing.

Comment (kvwalker) [reply test/functional/cursor_tests.js]: Here's the error: 
```
1) Cursor should apply parent read preference to count command:
     Uncaught TypeError: self.s.connectOptions.readPreference.equals is not a function
      at shouldTriggerConnect (/Users/katherinewalker/drivers/node/mongodb-core/lib/topologies/replset.js:770:42)
      at /Users/katherinewalker/drivers/node/mongodb-core/lib/topologies/replset.js:841:15
      at applyAuthenticationContexts (/Users/katherinewalker/drivers/node/mongodb-core/lib/topologies/replset.js:731:12)
      at Server.<anonymous> (/Users/katherinewalker/drivers/node/mongodb-core/lib/topologies/replset.js:802:7)
      at /Users/katherinewalker/drivers/node/mongodb-core/lib/topologies/server.js:506:16
      at /Users/katherinewalker/drivers/node/mongodb-core/lib/connection/pool.js:531:18
      at _combinedTickCallback (internal/process/next_tick.js:131:7)
      at process._tickCallback (internal/process/next_tick.js:180:9)
```

File: lib/mongo_client.js [hunk 2]
```diff
@@ -115,7 +115,6 @@ const validOptions = require('./operations/mongo_client_ops').validOptions;
  */
 function MongoClient(url, options) {
   if (!(this instanceof MongoClient)) return new MongoClient(url, options);
-
```
Comment (mbroadst) [inline lib/mongo_client.js]: nit: random whitespace change, let's try to keep these out of commits unless there are other useful changes.

**Comment by daprahamian** [comment]
I think this is worth backporting to the 3.0 branch. @mbroadst @kvwalker throughts?

**Comment by mbroadst** [comment]
@daprahamian agreed

**Comment by mbroadst** [comment]
@daprahamian agreed. Furthermore, the bug encountered with using `ReadPreference.SECONDARY` vs `ReadPreference.secondary` should be handled by the use of `resolveReadPreference`... I think we should investigate this further.

**Comment by mbroadst** [comment]
I made a ticket to track the `ReadPreference` in `MongoClient`s ctor [here](https://jira.mongodb.org/browse/NODE-1531)

**Pull request merged**